### ISSUE TITLE
It doesn't format the text when I run "Format document" (Alt + Shift + F)

### ISSUE THREAD

**Issue by fvogliotti** [DotJoshJohnson/vscode-xml#201]
#### Description
What seems to be the problem?
It doesn't format the text when I run "Format document" (Alt + Shift + F)
#### Formatter Implementation
Which XML Formatter implementation are you using (the value of your `xmlTools.xmlFormatterImplementation` setting).
v2
#### XML Tools Version
What version of XML Tools are you using?
2.3.1
#### VS Code Version
What version of VS Code are you using?
1.24.1
#### Operating System
What OS (and version) are you using?
Windows 10 Version 10.0.16299.431

**Comment by Stensby** [comment]
I am having the same issue using the new formatter.  XML Tools 2.3.1, VSCode 1.24.1, Ubuntu 17.10.  I can see the extension handler process using a lot of CPU but nothing is formatted (even after waiting 10+ seconds).

Switching back to "classic" and the formatter works on my xml file within 2 seconds.

**Comment by fvogliotti** [comment]
I started using classic mode but it leaves a lot of spaces in large xml documents. The good is that it works but the bad is that leaves a lot of spaces. Sometimes you have to scroll far to the right to see text.
Other than that it's a really great extension with a lot of use!

**Comment by DotJoshJohnson** [comment]
Can someone provide a sample of some XML that causes this behavior?

**Comment by fvogliotti** [comment]
Hi Josh, I'll see if I can send you a copy of the file. Thanks for your support!
________________________________
De: Josh Johnson <notifications@github.com>
Enviado: sábado, 30 de junio de 2018 09:55
Para: DotJoshJohnson/vscode-xml
Cc: Fran; Author
Asunto: Re: [DotJoshJohnson/vscode-xml] It doesn't format the text when I run "Format document" (Alt + Shift + F) (#201)


Can someone provide a sample of some XML that causes this behavior?

**Comment by fappaz** [comment]
Hi there,

I'm having the exact same problem as reported by @frvo. When Format Document is executed in an XML file, nothing happens. It seems to be an issue just with fairly long XML files (containing thousands of entries) and I managed to reproduce it with the attached test file. 

[xml-issue201.txt](https://github.com/DotJoshJohnson/vscode-xml/files/2276011/xml-issue201.txt)

This can be reproduced when there are no other plugins enabled. Nothing is logged in the console when the Developer Tools are toggled on.

The same file can be formatted properly using Notepad++ v7.5.4 (32 bit) and XML Tools Plugin 2.4.8.

My setup:
- Windows 10 Home 64 bits
- VSCode 1.25.1
- XML Tools 2.3.1
- "xmlTools.xmlFormatterImplementation": "v2"

**Comment by fappaz** [comment]
To make it even more weird, I opened the Developer Tools in VSCode and clicked on the X button to close it. The entire app froze for about 2 seconds and shutdown. When I opened it again, I had to import the `xml-issue201.txt` file (that I attached in the previous message), switched the language mode to XML and formatted it - and this time it worked!
However, when I tried to format the real file, which has about 1.6k lines, the problem persists.

**Comment by DotJoshJohnson** [comment]
@fappaz - VS Code by default uses file extensions to identify XML files. Using the native "format document" shortcuts will only work if VS Code identifies the file as an XML document. The `XML Tools: Format as XML Document` command, however, is ALWAYS available for any file type. I tested using your provided txt file and was able to format the file using the "format as XML" feature.

@frvo and @Stensby - Your issues are likely related to the structure of the XML document. The formatter currently does a pretty poor job of reporting issues if an error is thrown. It just dies silently. If you can provide a copy of a file you are having trouble with, I mkay be able to track down what the formatter is choking on.

I'm closing this for now, but will re-open if sample files are provided.

**Comment by fvogliotti** [comment]
Ok! Thank you Josh! To be honest with you, I'm working with xml files that
are part of a company and I'm really not sure if I am allowed to submit
that kind of files. I tried to search for other type of files to send you
but I didn't manage to get the error reproduced. So I'm very sorry about
this. As a developer for a large company I'm really eager to recommend your
extension as well as the use of VSCode, in fact I did with several of my
colleagues here. When I work with these large XMLs what I do to get around
this problem is to copy and paste the portion of text that I need and work
with it with no problems.
Again, thank you very much for your support and for giving the community
such a great tool to work with. You made our days (and daily tasks) so
much easier and enjoyable.
Thanks,
Fran
